parachain/pallets/ethereum-light-client/src/lib.rs

```diff
@@ -265,6 +265,48 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		/// Reset's the internal state of the light client.
+		///
+		/// `forked_at` - The hash of the header where the fork originated. The common ancestor.
+		///
+		/// This is only meant to be used in staging environments where long range forks
+		/// can happen.
+		///
+		/// Requires sudo user.
+		#[pallet::weight(1_000_000)]
+		#[transactional]
+		pub fn handle_long_range_fork(origin: OriginFor<T>, forked_at: H256) -> DispatchResult {
+			ensure_root(origin)?;
+
+			let stored_header = <Headers<T>>::get(forked_at).ok_or(Error::<T>::MissingHeader)?;
+			let header = stored_header.header;
```

Comment:
Nit: Should be able to collapse to:

```suggestion
			let header = <Headers<T>>::get(forked_at).ok_or(Error::<T>::MissingHeader)?.header;
```

Should compile